vm/src/api.rs

```diff
@@ -1320,6 +1320,36 @@ macro_rules! record {
     }
 }
 
+#[macro_export]
+macro_rules! hlist_p {
```

Comment:
Decided to hide the internal macros inside the macros themselves since they are only used in one macro each.